### PR TITLE
Retarget .NET framework Octokit.GraphQL and Octokit.GraphQL.Core

### PR DESCRIPTION
**Pull request by onein528**
## Description

As the title

## Motivation & Context

Related: #262

### CODE REVIEW

**Comment by jeffhandley** [comment]
It looks like all projects in the solution will need be updated to target netstandard2.0, including the unit test projects.

**Comment by onein528** [comment]
Entire solution?
Okay. I'll be change.

**Comment by onein528** [comment]
CI is configured for .NET Core and I don't know hot to re-configure.

@nickfloyd How do you feel about this PR?

File: Octokit.GraphQL.Core.Generation.UnitTests/Octokit.GraphQL.Core.Generation.UnitTests.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (jeffhandley) [inline Octokit.GraphQL.Core.Generation.UnitTests/Octokit.GraphQL.Core.Generation.UnitTests.csproj:3]: Sorry, my mistake in my previous comment. For projects that were targeting `netcoreapp`, those should stay as-is. But for projects that were targeting `netstandard1.1`, those should be updated to `netstandard2.0`.

Only class libraries can target `netstandard`; unit tests and other apps need to target a `netcoreapp`.

```suggestion
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

File: Octokit.GraphQL.Core.UnitTests/Octokit.GraphQL.Core.UnitTests.csproj [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (jeffhandley) [inline Octokit.GraphQL.Core.UnitTests/Octokit.GraphQL.Core.UnitTests.csproj:3]:
```suggestion
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

File: Octokit.GraphQL.IntegrationTests/Octokit.GraphQL.IntegrationTests.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (jeffhandley) [inline Octokit.GraphQL.IntegrationTests/Octokit.GraphQL.IntegrationTests.csproj:3]:
```suggestion
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

File: Octokit.GraphQL.UnitTests/Octokit.GraphQL.UnitTests.csproj [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (jeffhandley) [inline Octokit.GraphQL.UnitTests/Octokit.GraphQL.UnitTests.csproj:3]:
```suggestion
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

File: Tools/Generate/Generate.csproj [hunk 5]
```diff
@@ -1,10 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (jeffhandley) [inline Tools/Generate/Generate.csproj:4]:
```suggestion
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

File: Octokit.GraphQL.nuspec [hunk 6]
```diff
@@ -18,9 +18,9 @@
...
   </metadata>
   <files>
-    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.dll" target="lib\netstandard1.1"/>
-    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.xml" target="lib\netstandard1.1"/>
-    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.Core.dll" target="lib\netstandard1.1"/>
-    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.Core.xml" target="lib\netstandard1.1"/>
+    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.dll" target="lib\netstandard2.0"/>
+    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.xml" target="lib\netstandard2.0"/>
+    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.Core.dll" target="lib\netstandard2.0"/>
+    <file src="Octokit.GraphQL\bin\Release\netstandard1.1\Octokit.GraphQL.Core.xml" target="lib\netstandard2.0"/>
```
Comment (jeffhandley) [inline Octokit.GraphQL.nuspec:24]: These `src` paths need to be updated too: (I overlooked that before)

```suggestion
    <file src="Octokit.GraphQL\bin\Release\netstandard2.0\Octokit.GraphQL.dll" target="lib\netstandard2.0"/>
    <file src="Octokit.GraphQL\bin\Release\netstandard2.0\Octokit.GraphQL.xml" target="lib\netstandard2.0"/>
    <file src="Octokit.GraphQL\bin\Release\netstandard2.0\Octokit.GraphQL.Core.dll" target="lib\netstandard2.0"/>
    <file src="Octokit.GraphQL\bin\Release\netstandard2.0\Octokit.GraphQL.Core.xml" target="lib\netstandard2.0"/>
```

**Comment by jeffhandley** [comment]
> All checks have passed

:tada: Nice job, @onein528.

@nickfloyd What do you think? If we can get a new version published with this retargeting from netstandard1.1 to netstandard2.0, that would help me on my current project. I am building a .NET Console app hosted in a docker container, serving as a GitHub Action. With this fix, I'll be able to use Native AOT to produce a very small image with a self-contained exe that uses GitHub GraphQL APIs through Octokit. I think that'll get my action response times down below 10 seconds consistently. 😃

**Comment by onein528** [comment]
Niceee!!!

**Comment by nickfloyd** [comment]
@onein528 & @jeffhandley, thanks for this change.  Great timing, given we are wanting to do the same with [octokit.net](https://github.com/octokit/octokit.net/issues/2500) ❤️ 

I've marked it as a breaking change, so we can bump the major semver and highlight this change in the release note. I'd like to get this released on Monday (08/01/2022) and follow up with a new schema generation.

Let me know if that makes sense - and I'll plan on getting it out the door!

**Comment by jeffhandley** [comment]
That's perfect for us, @nickfloyd! Thank you. If you publish a prerelease package to NuGet, I'd love to test drive it for you and verify that AOT works for my scenarios.

**Review by nickfloyd — APPROVED** [review]
🎉

**Comment by nickfloyd** [comment]
Hey friends, the [release](https://github.com/octokit/octokit.graphql.net/releases/tag/v0.2.0-beta) is out on [nuget](https://www.nuget.org/packages/Octokit.GraphQL/) - let me know if you run into anything!  Thanks again for helping to get this done and across the finish line!

**Comment by jeffhandley** [comment]
Thanks, @nickfloyd! I've bumped from `0.1.9-beta` to `0.2.0-beta` in my project, and I can now successfully use .NET AOT. 🎉 

* Commit with the update: https://github.com/jeffhandley/github-actions/commit/5418ee2afbb9dd8bab1fd52e3660747609f87343
* Successful GitHub action run: https://github.com/jeffhandley/github-action-testing/runs/7637969596?check_suite_focus=true

**Pull request merged**